bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -591,7 +602,7 @@ impl Connection {
 #[derive(Default)]
 struct Context {
     waker: Option<Waker>,
-    pending_callback: Option<Box<dyn AsyncCallback>>,
+    client_hello_future: Option<Pin<Box<dyn ConnectionFuture>>>,
```

Comment:
It looks like you're replacing a solution that supports all async callbacks with a solution that only supports client hello callbacks. Can this be extended? Does it have to only work for the client hello callback?

Reply:
The previous solution also only worked for client_hello since `AsyncClientHelloCallback` did stuff specific to client hello. I simply replaced `AsyncClientHelloCallback` with `poll_client_hello_callback` since we no longer need to construct an `AsyncCallback` object to store on the connection.

Reply:
So we can still support other async callbacks like previously. We just need a custom impl of `poll_client_hello_callback` -> `poll_x_callback` and need to store the future on the connection, which is what we would have needed previously also.

Reply:
AsyncClientHelloCallback did stuff specific to the client hello callback, but pending_callback could be set to any AsyncCallback. You're just storing a "ConnectionFuture" here-- why does that ConnectionFuture have to specific to the client hello callback?